site/vue/src/components/popup.vue

```diff
@@ -0,0 +1,119 @@
+<script setup lang="ts">
+import { watch } from 'vue';
+
+const props = withDefaults(defineProps<{
+    id?: string;
+    visible?: boolean;
+    closeButton?: {
+        id: string;
+        classes: string[];
+    };
+}>(), {
+    id: 'popup',
+    title: 'popup title',
+    closeButton() {
+        return {
+            id: 'close_button',
+            classes: ['btn-default'],
+        };
+    },
+
+});
+
+const emit = defineEmits<{
+    dismiss: [];
+}>();
+
+watch(
+    () => props.visible,
+    (visible, old_visible) => {
+        if (old_visible !== visible) {
+            if (visible) {
+                $(document).on(
+                    'keydown.popup',
+                    (event) => {
+                        if (event.key === 'Escape' && props.visible) {
+                            emit('dismiss');
+                        }
+                    });
+            }
+            else {
+                $(document).off('keydown.popup');
+            }
+        }
+    },
+    { immediate: true },
+);
+
+</script>
+
+<template>
+  <Teleport
```

Comment:
How hard would it be to avoid using Teleport here?  A messy DOM structure like this will be pretty confusing for new developers, and incurs a significant amount of technical debt.

Reply:
I genutively thought it would be cleaner :(. It's not nessesary, reverted in https://github.com/Submitty/Submitty/pull/10782/commits/e7c1aa4ed6c4b5070bc5419b48f6e26ec3ce61b5